app/queries/affected_by_recurring_blackout_query.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AffectedByRecurringBlackoutQuery < QueryBase
```

Comment:
I think this can end up living in `app/queries/reservations` and inherit from `ReservationsQueryBase` - in the interest of having query objects that can be chained, it makes sense to allow for the base query / relation to be overridden.